### PR TITLE
DAOS-4928 dfuse: Remove inode record table from dfuse.

### PR DESCRIPTION
**Pull request by ashleypittman**
Overhaul how inodes are allocated in dfuse, rather than allocating
an integer and searching that it's unique make a inode from the
oid and dfs, and then check that it's unique.

This removes the inode record table entirely, which was a record
of every inode used ever, where all lookups were negative and
uses just the inode entry table which was searched anyway, there's
still a find_insert() call for each inode here.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Comment by ashleypittman** [comment]
I've tidied this up and it's ready for review now.

File: src/client/dfuse/dfuse_core.c [hunk 1]
```diff
@@ -284,17 +221,12 @@ dfuse_start(struct dfuse_info *dfuse_info, struct dfuse_dfs *dfs)
 	fs_handle->dpi_max_read = 1024 * 1024 * 4;
 	fs_handle->dpi_max_write = 1024 * 1024;
 
-	rc = d_hash_table_create_inplace(D_HASH_FT_RWLOCK | D_HASH_FT_EPHEMERAL,
-					 3, fs_handle, &ie_hops,
+	rc = d_hash_table_create_inplace(D_HASH_FT_MUTEX | D_HASH_FT_LRU | D_HASH_FT_EPHEMERAL,
```
Comment (jolivier23) [inline src/client/dfuse/dfuse_core.c:224]: we should really either enforce the 80 character rule where possible or change checkpatch to something like 120 and enforce that.

**Review by jolivier23 — APPROVED** [review]

File: src/client/dfuse/dfuse_core.c (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfuse/dfuse_core.c:224]: Agree, I think this is a case where an exemption is warranted, but I think we should change it to 120.

**Review by johannlombardi — APPROVED** [review]

**Comment by ashleypittman** [comment]
Failures look to be DAOS-6090 and DAOS-6017 and DAOS-6061

**Pull request merged**